### ISSUE TITLE
[Feature]: Add --prev or --previous flag to `uplift tag`

### ISSUE THREAD

**Issue by nashikb** [gembaadvantage/uplift#244]
### Describe your feature

There are use cases where I/we want to know what the current semantic version is as well as the /next/ version. It's useful to know if a repo went from v1.3.0 to v1.3.1 vs v1.3.0 to v1.4 or v2.0, etc, in CI/CD tooling. The output of `changelog` exposes the "current" tag, but again it's human-readable stderr, or a parse of a changelog diff.

### Your potential solution

While the documentation states that the `uplift tag --next` command emits the current tag to stderr, this does not seem to be the case in the current version. Additionally, potentially parsing this current version out of human-readable stderr seems unreliable.

Would it be possible to add a separate `--prev` or `--previous` flag to the `tag` command that would emit the most recent semantic version? Alternately, a new `--state` or somesuch flag that would emit both the current and would-be-created new semantic version?

### Any additional information?

Thanks!

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by ga-paul-t** [comment]
👋 @nashikb, thanks for getting in touch.

I will take a look at the issue with regard to the `upilft tag --next` command. 

I wanted to ask a quick question about the proposal of a `--state` flag. What use cases do you envisage using such a command to print out both the existing and next calculated tags? Could the output be as simple as `current:next` or did you have a format in mind? Are you planning to work on this output within your CI?

**Comment by nashikb** [comment]
My thought on `--state` was from the perspective of having one command that provides both numbers, so CI doesn't necessarily have to call uplift twice. On further reflection, maybe allowing `--next` /and/ a new `--current` option at the same time could be useful, rather than a net-new flag, and would maintain backwards compatibility? The `:` separator might be slightly confusing given Docker's use of `:` to mark versions; perhaps simple space delimited? Or perhaps `..` like git?

In my use case, knowing the current version allows me to know which docker image to pull for a given repo state, what message to put in various other metadata, etc. 

Hopefully that helps?

Thanks!

**Comment by ga-paul-t** [comment]
@nashikb, thanks for the response. That definitely helps

**Comment by ga-paul-t** [comment]
Morning @nashikb 

I would like to get your feedback on the proposed change I have made. So I have introduced a new `--current` flag to the `uplift tag` command that will effectively short circuit the existing tag workflow and print the latest semantic repository tag to stdout. Existing behaviour when providing the `--next` tag is still intact. So ultimately you get the following three scenarios:

1. `uplift tag --current`: will print the current semantic version to stdout e.g. `v0.1.0`. If no tag is identified, then nothing is printed
2. `uplift tag --next`: will print the next calculated semantic version to stdout e.g. `v0.2.0`. This uses the existing uplift workflow, which ensures backwards compatibility. If the current commit doesn't produce a new semantic version, then nothing is printed to stdout
3. `uplift tag --current --next`: will print the current and next calculated semantic version to stdout e.g. `v0.1.0 v0.2.0`. The output is space delimited, ensuring compatibility with existing Linux tools such as `cut`. As this triggers the existing workflow, if no new semantic version is calculated, then nothing is printed to stdout

If `3.` was to still print out the current tag if no next tag was present, the existing workflow would need to be refactored which is a much bigger change
